bgpd/bgp_conditional_adv.h

```diff
@@ -0,0 +1,130 @@
+/*
+ * BGP Conditional advertisement
+ * Copyright (C) 2020  Samsung Research Institute Bangalore.
+ * 		       Madhurilatha Kuruganti
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the Free
+ * Software Foundation; either version 2 of the License, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; see the file COPYING; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA
+ */
+
+#ifndef _FRR_BGP_CONDITION_ADV_H
+#define _FRR_BGP_CONDITION_ADV_H
+#include <zebra.h>
+#include "prefix.h"
+#include "bgpd/bgp_addpath.h"
+#include "bgpd/bgp_attr.h"
+#include "bgpd/bgpd.h"
+#include "bgpd/bgp_debug.h"
+#include "bgpd/bgp_route.h"
+#include "bgpd/bgp_updgrp.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/* Polling time for monitoring condition-map routes in route table */
+#define CONDITIONAL_ROUTES_POLL_TIME 60
+
+#define FOREACH_ACCESS_LIST_FILTER(alist, filter)                              \
+	for (filter = alist->head; filter; filter = filter->next)
+
+static inline bool is_rmap_valid(struct route_map *rmap)
+{
+	if (!rmap || !rmap->head)
+		return false;
+
+	/* Doesn't make sense to configure advertise
+	 * or condition map in deny/any clause. */
+	if (!rmap->head->type == RMAP_PERMIT)
```

Comment:
OpenBSD compiler is rightfully complaining here: do we want to negate `rmap->head->type` or the condition `rmap->head->type == RMAP_PERMIT`?

Please use parentheses to disambiguate.

Reply:
its if (rmap->head->type != RMAP_PERMIT)

Reply:
this function is removed after added code to support other match rules to conditional advertisement

Reply:
used the existing route_map_apply()